internal/rollup/uglify-wrapped.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * @license
+ * Copyright 2017 The Bazel Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// A wrapper around uglify-js that handle minifies directories as inputs
+// and outputs
+
+const fs = require('fs');
+const path = require('path');
+const child_process = require('child_process');
+
+const DEBUG = false;
+
+// capture the inputs and output options
+const argv = require('minimist')(process.argv.slice(2));
+const inputs = argv._;
+const output = argv.output || argv.o;
+const debug = argv.debug;
+const configFile = argv['config-file'];
+delete argv._;
+delete argv.output;
+delete argv.o;
+delete argv.debug;
+delete argv['config-file'];
+
+const isWindows = /^win/i.test(process.platform);
+
+if (DEBUG)
+  console.error(`
+Uglify: running with
+  cwd: ${process.cwd()}
+  argv: ${process.argv.slice(2).join(' ')}
+  inputs: ${JSON.stringify(inputs)}
+  output: ${output}
+  debug: ${debug}
+`);
+
+if (inputs.length != 1) {
+  throw new Error(`Only one input file supported: ${inputs}`);
+}
+
+const input = inputs[0];
+
+function runUglify(inputFile, outputFile, sourceMapFile) {
+  if (DEBUG) console.error(`Minifying ${inputFile} -> ${outputFile} (sourceMap ${sourceMapFile})`);
+
+  const uglifyConfig = {
```

Comment:
is there actually something more dynamic about this now, that required moving it from a template file?

Reply:
yes. uglify may be called once for the non-code-split configuration but for the code-split case it is called multiple times and the config files is actually written out multiple times during the execution of the script:

```
if (isFile) {
  runUglify(input, output, output + '.map');
} else {
  if (!fs.existsSync(output)) {
    fs.mkdirSync(output);
  }
  const dir = fs.readdirSync(input);
  dir.forEach(f => {
    if (f.endsWith('.js')) {
      const inputFile = path.join(input, path.basename(f));
      const outputFile = path.join(output, path.basename(f));
      runUglify(inputFile, outputFile, outputFile + '.map');
    }
  });
}
```

Come to think of it. Since the config file name comes from outside of the script it is odd that in the code-split case that one file will be overwritten multiple times. A tmp file should probably be used and the non-code split case could be parallelized.

Reply:
Add a TODO for parallelizing it, if that's something we intend to do? I assume it would still be one Bazel action, but would shell out to multiple nodejs processes to run several uglify tasks?